src/org/mozilla/javascript/Kit.java

```diff
@@ -34,20 +73,22 @@
         } catch (Exception ex) {
             // Assume any exceptions means the method does not exist.
         }
+
+        reinitNonClassCache();
+
+        //identify enabling of non-class cache
+        try {
+            enabledNonClassCaching = CACHE_USE_PROPERTY_NAME_YES.equals(System.getProperty(CACHE_USE_PROPERTY_NAME));
```

Comment:
Config settings like this should be exposed through a feature flag in the Context class (see https://github.com/mozilla/rhino/pull/619#discussion_r344956743 for example